### DISCUSSION TITLE
Source code SSH settings from local .ssh/config

### DISCUSSION THREAD

**Discussion by msbit** [deployphp/deployer#3826] [Help needed]
- Deployer version: v7.4.0
- Deployment OS: macOS

I have a a repository set via something like:

```
set('repository', 'ssh://code.foo.example.com/repo');
```

a host set up something like:

```
host('target.example.com')
    ->set('remote_user', 'user')
    ->set('deploy_path', '~/repo');
```

and per-user SSH settings configured for this repository on my local machine, in `.ssh/config`, something like:

```
Host code.*.example.com
  Ciphers       aes256-ctr
  IdentityFile  ~/.ssh/example_rsa
  KexAlgorithms diffie-hellman-group-exchange-sha256
  User          different-user
```

These settings are not used when attempting to fetch the repo during the `deploy:update_code` step, so this step fails.

What can I do to achieve this?

**Comment by antonmedv** [comment]
>  on my local machine

`deploy:update_code` fetches on remote machine. Configure .ssh/config on remote machine.

**Comment by msbit** [comment]
Thanks @antonmedv, so if I do so, can ssh agent forwarding be setup such that I can use my local `IdentityFile`?
